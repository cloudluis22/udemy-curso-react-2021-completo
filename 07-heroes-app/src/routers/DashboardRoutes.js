import React from 'react'
import {

  BrowserRouter as Router,
  Switch,
  Route,
  Redirect

} from "react-router-dom";
import { DCScreen } from '../components/dc/DCScreen';
import { HeroScreen } from '../components/heroes/HeroScreen';
import { MarvelScreen } from '../components/marvel/MarvelScreen';
import { SearchScreen } from '../components/search/SearchScreen';
import { Navbar } from '../components/ui/NavBar';

export const DashboardRoutes = () => {
    return (
        <>
            <Navbar />

            <div className="container mt-2">
                <Switch>

                    <Route exact path="/marvel" component={MarvelScreen} />
                    <Route exact path="/hero/:heroId" component={HeroScreen} />
                    <Route exact path="/dc" component={DCScreen} />
                    <Route exact path="/search" component={SearchScreen} />

                    <Redirect to="/marvel" />
                </Switch>
            </div>


        </>
    )
}
