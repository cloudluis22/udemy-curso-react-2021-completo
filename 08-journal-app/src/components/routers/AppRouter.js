import React, { useEffect, useState } from 'react'
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect
} from "react-router-dom";
import { firebase } from '../../firebase/firebaseConfig';
import { useDispatch } from 'react-redux';

import AuthRouter from './AuthRouter';
import JournalScreen from '../journal/JournalScreen';
import { login } from '../../actions/auth';
import { PublicRoute } from './PublicRoute';
import { PrivateRoute } from './PrivateRoute';
import { startLoadingNotes } from '../../actions/notes'

const AppRouter = () => {

    const dispatch = useDispatch();
    const [checking, setChecking] = useState(true);
    const [isLoggedIn, setIsLoggedIn] = useState(false);

    console.log(isLoggedIn);

    useEffect( () => {

        firebase.auth().onAuthStateChanged( async (user) => {

            if (user?.uid) {
                dispatch(login(user.uid, user.displayName));
                setIsLoggedIn(true);

                dispatch( startLoadingNotes(user.uid) );
            }
            else {
                setIsLoggedIn(false);
            }

            // Una vez se autentica ya no se está revisando si el usuario está siendo autenticado.
            setChecking(false)
        })

    }, [dispatch, setChecking, setIsLoggedIn]);

    if (checking) {
        return (
            <div className="spinner round"></div>
        )
    }

    return (
        <Router>
            <div>
                <Switch>

                    <PublicRoute isLoggedIn={ isLoggedIn } component={ AuthRouter } path="/auth" />
                    <PrivateRoute exact isLoggedIn={ isLoggedIn } path="/" component={ JournalScreen } />

                    <Redirect to="/auth/login" />

                </Switch>
            </div>
        </Router>
    )
}

export default AppRouter;