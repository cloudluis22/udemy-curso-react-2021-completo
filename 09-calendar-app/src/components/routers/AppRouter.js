import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import {
  BrowserRouter as Router,
  Switch,
  Redirect
} from "react-router-dom";
import { startChecking } from '../../actions/auth';
import { LoginScreen } from '../auth/LoginScreen';
import { CalendarScreen } from '../calendar/CalendarScreen';
import { PrivateRoute } from './PrivateRoute';
import { PublicRoute } from './PublicRoute';

export const AppRouter = () => {

    const dispatch = useDispatch();
    const { checking, uid } = useSelector(state => state.auth);

    useEffect(() => {
        dispatch( startChecking() );
    }, [dispatch]);

    if (checking) {
        return <h5> Espere... </h5>
    }

    return (
        <Router>
            <div>
                <Switch>

                    <PublicRoute exact path="/login" isLoggedIn={ !!uid } component={ LoginScreen } />
                    <PrivateRoute exact path="/" isLoggedIn={ !!uid  } component={ CalendarScreen } />

                    <Redirect to="/" />

                </Switch>
            </div>
        </Router>
    )
}
