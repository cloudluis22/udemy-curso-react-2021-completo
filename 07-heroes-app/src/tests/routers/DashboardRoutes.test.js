import React from 'react';
import { mount } from 'enzyme';
import { DashboardRoutes } from '../../routers/DashboardRoutes';
import AuthContext from '../../auth/AuthContext';
import { MemoryRouter } from 'react-router-dom';

describe('Pruebas en el componente de "<DashboardRoutes />."', () => {

    test('Debe de mostrarse correctamente.', () => {

        const contextValue = {
            dispatch: jest.fn(),
            user: {
                logged: true,
                name: 'Hola'
            }
        }

        const wrapper = mount(

            <AuthContext.Provider value={contextValue}>

                <MemoryRouter>

                    <DashboardRoutes />

                </MemoryRouter>

            </AuthContext.Provider>

        );

        expect(wrapper).toMatchSnapshot();

    });

});