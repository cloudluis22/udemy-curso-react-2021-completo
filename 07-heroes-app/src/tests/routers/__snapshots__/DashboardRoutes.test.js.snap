// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente de "<DashboardRoutes />." Debe de mostrarse correctamente. 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/marvel"
          onClick={[Function]}
          style={Object {}}
        >
          Marvel
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div>
      <ul
        className="navbar-nav ml-auto px-5"
      >
        <span
          className="nav-item nav-link text-info"
        >
          Hola
        </span>
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container mt-2"
  >
    <div>
      <h1>
        Marvel Screen
      </h1>
      <hr />
      <div
        className="row row-cols-1 row-cols-md-3 animate__animated animate__fadeInUp animate__fast"
      >
        <div
          className="card mb-5 ms-5 pe-1 "
          style={
            Object {
              "maxWidth": 350,
            }
          }
        >
          <img
            alt="hero"
            className="card-img-top mt-3 rounded mx-auto d-block"
            src="../assets/heroes/marvel-spider.jpg"
            style={
              Object {
                "maxHeight": 300,
                "maxWidth": 250,
              }
            }
          />
          <div
            className="card-body"
          >
            <h4
              className="card-title"
            >
               
              Spider Man
               
            </h4>
            <h5
              className="card-subtitle text-muted"
            >
               
              Peter Parker
               
            </h5>
            <a
              href="/hero/marvel-spider"
              onClick={[Function]}
            >
              ...Más
            </a>
          </div>
        </div>
        <div
          className="card mb-5 ms-5 pe-1 "
          style={
            Object {
              "maxWidth": 350,
            }
          }
        >
          <img
            alt="hero"
            className="card-img-top mt-3 rounded mx-auto d-block"
            src="../assets/heroes/marvel-captain.jpg"
            style={
              Object {
                "maxHeight": 300,
                "maxWidth": 250,
              }
            }
          />
          <div
            className="card-body"
          >
            <h4
              className="card-title"
            >
               
              Captain America
               
            </h4>
            <h5
              className="card-subtitle text-muted"
            >
               
              Steve Rogers
               
            </h5>
            <a
              href="/hero/marvel-captain"
              onClick={[Function]}
            >
              ...Más
            </a>
          </div>
        </div>
        <div
          className="card mb-5 ms-5 pe-1 "
          style={
            Object {
              "maxWidth": 350,
            }
          }
        >
          <img
            alt="hero"
            className="card-img-top mt-3 rounded mx-auto d-block"
            src="../assets/heroes/marvel-iron.jpg"
            style={
              Object {
                "maxHeight": 300,
                "maxWidth": 250,
              }
            }
          />
          <div
            className="card-body"
          >
            <h4
              className="card-title"
            >
               
              Iron Man
               
            </h4>
            <h5
              className="card-subtitle text-muted"
            >
               
              Tony Stark
               
            </h5>
            <a
              href="/hero/marvel-iron"
              onClick={[Function]}
            >
              ...Más
            </a>
          </div>
        </div>
        <div
          className="card mb-5 ms-5 pe-1 "
          style={
            Object {
              "maxWidth": 350,
            }
          }
        >
          <img
            alt="hero"
            className="card-img-top mt-3 rounded mx-auto d-block"
            src="../assets/heroes/marvel-thor.jpg"
            style={
              Object {
                "maxHeight": 300,
                "maxWidth": 250,
              }
            }
          />
          <div
            className="card-body"
          >
            <h4
              className="card-title"
            >
               
              Thor
               
            </h4>
            <h5
              className="card-subtitle text-muted"
            >
               
              Thor Odinson
               
            </h5>
            <a
              href="/hero/marvel-thor"
              onClick={[Function]}
            >
              ...Más
            </a>
          </div>
        </div>
        <div
          className="card mb-5 ms-5 pe-1 "
          style={
            Object {
              "maxWidth": 350,
            }
          }
        >
          <img
            alt="hero"
            className="card-img-top mt-3 rounded mx-auto d-block"
            src="../assets/heroes/marvel-hulk.jpg"
            style={
              Object {
                "maxHeight": 300,
                "maxWidth": 250,
              }
            }
          />
          <div
            className="card-body"
          >
            <h4
              className="card-title"
            >
               
              Hulk
               
            </h4>
            <h5
              className="card-subtitle text-muted"
            >
               
              Bruce Banner
               
            </h5>
            <a
              href="/hero/marvel-hulk"
              onClick={[Function]}
            >
              ...Más
            </a>
          </div>
        </div>
        <div
          className="card mb-5 ms-5 pe-1 "
          style={
            Object {
              "maxWidth": 350,
            }
          }
        >
          <img
            alt="hero"
            className="card-img-top mt-3 rounded mx-auto d-block"
            src="../assets/heroes/marvel-wolverine.jpg"
            style={
              Object {
                "maxHeight": 300,
                "maxWidth": 250,
              }
            }
          />
          <div
            className="card-body"
          >
            <h4
              className="card-title"
            >
               
              Wolverine
               
            </h4>
            <h5
              className="card-subtitle text-muted"
            >
               
              James Howlett
               
            </h5>
            <a
              href="/hero/marvel-wolverine"
              onClick={[Function]}
            >
              ...Más
            </a>
          </div>
        </div>
        <div
          className="card mb-5 ms-5 pe-1 "
          style={
            Object {
              "maxWidth": 350,
            }
          }
        >
          <img
            alt="hero"
            className="card-img-top mt-3 rounded mx-auto d-block"
            src="../assets/heroes/marvel-daredevil.jpg"
            style={
              Object {
                "maxHeight": 300,
                "maxWidth": 250,
              }
            }
          />
          <div
            className="card-body"
          >
            <h4
              className="card-title"
            >
               
              Daredevil
               
            </h4>
            <h5
              className="card-subtitle text-muted"
            >
               
              Matthew Michael Murdock
               
            </h5>
            <a
              href="/hero/marvel-daredevil"
              onClick={[Function]}
            >
              ...Más
            </a>
          </div>
        </div>
        <div
          className="card mb-5 ms-5 pe-1 "
          style={
            Object {
              "maxWidth": 350,
            }
          }
        >
          <img
            alt="hero"
            className="card-img-top mt-3 rounded mx-auto d-block"
            src="../assets/heroes/marvel-hawkeye.jpg"
            style={
              Object {
                "maxHeight": 300,
                "maxWidth": 250,
              }
            }
          />
          <div
            className="card-body"
          >
            <h4
              className="card-title"
            >
               
              Hawkeye
               
            </h4>
            <h5
              className="card-subtitle text-muted"
            >
               
              Clinton Francis Barton
               
            </h5>
            <a
              href="/hero/marvel-hawkeye"
              onClick={[Function]}
            >
              ...Más
            </a>
          </div>
        </div>
        <div
          className="card mb-5 ms-5 pe-1 "
          style={
            Object {
              "maxWidth": 350,
            }
          }
        >
          <img
            alt="hero"
            className="card-img-top mt-3 rounded mx-auto d-block"
            src="../assets/heroes/marvel-cyclops.jpg"
            style={
              Object {
                "maxHeight": 300,
                "maxWidth": 250,
              }
            }
          />
          <div
            className="card-body"
          >
            <h4
              className="card-title"
            >
               
              Cyclops
               
            </h4>
            <h5
              className="card-subtitle text-muted"
            >
               
              Scott Summers
               
            </h5>
            <a
              href="/hero/marvel-cyclops"
              onClick={[Function]}
            >
              ...Más
            </a>
          </div>
        </div>
        <div
          className="card mb-5 ms-5 pe-1 "
          style={
            Object {
              "maxWidth": 350,
            }
          }
        >
          <img
            alt="hero"
            className="card-img-top mt-3 rounded mx-auto d-block"
            src="../assets/heroes/marvel-silver.jpg"
            style={
              Object {
                "maxHeight": 300,
                "maxWidth": 250,
              }
            }
          />
          <div
            className="card-body"
          >
            <h4
              className="card-title"
            >
               
              Silver Surfer
               
            </h4>
            <h5
              className="card-subtitle text-muted"
            >
               
              Norrin Radd
               
            </h5>
            <a
              href="/hero/marvel-silver"
              onClick={[Function]}
            >
              ...Más
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
